packages/remix-react/single-fetch.tsx

```diff
@@ -134,26 +134,31 @@ export function getSingleFetchDataStrategy(
       // 2. if multiple call `serverLoader` only one fetch call is made
       let singleFetchPromise: Promise<SingleFetchResults>;
 
-      let makeSingleFetchCall = async () => {
-        let url = addRevalidationParam(
-          manifest,
-          routeModules,
-          matches.map((m) => m.route),
-          matches.filter((m) => m.shouldLoad).map((m) => m.route),
-          // Single fetch doesn't need/want naked index queries on action
-          // revalidation requests
-          stripIndexParam(singleFetchUrl(request.url))
-        );
-
-        let { data } = await fetchAndDecode(url);
-        return data as SingleFetchResults;
-      };
-
       singleFetch = async (routeId) => {
-        if (!singleFetchPromise) {
-          singleFetchPromise = makeSingleFetchCall();
+        let results: SingleFetchResults;
+        if (manifest.routes[routeId].hasClientLoader) {
+          // When a route has a client loader, we make it's own call for just
+          // it's server loader data
+          let url = stripIndexParam(singleFetchUrl(request.url));
+          url.searchParams.set("_routes", routeId);
+          let { data } = await fetchAndDecode(url);
+          results = data as SingleFetchResults;
```

Comment:
`clientLoader` calls through to `serverLoader` make a call for their individual route